cmd/bootstrap/cmd/clusters.go

```diff
@@ -8,45 +8,40 @@ import (
 	"github.com/onflow/flow-go/model/flow/assignment"
 	"github.com/onflow/flow-go/model/flow/factory"
 	"github.com/onflow/flow-go/model/flow/filter"
+	"github.com/onflow/flow-go/utils/rand"
 )
 
-// Construct cluster assignment with internal and partner nodes uniformly
-// distributed across clusters. This function will produce the same cluster
-// assignments for the same partner and internal lists, and the same seed.
-func constructClusterAssignment(partnerNodes, internalNodes []model.NodeInfo, seed int64) (flow.AssignmentList, flow.ClusterList) {
-
-	partners := model.ToIdentityList(partnerNodes).Filter(filter.HasRole(flow.RoleCollection))
-	internals := model.ToIdentityList(internalNodes).Filter(filter.HasRole(flow.RoleCollection))
-	nClusters := flagCollectionClusters
-	nCollectors := len(partners) + len(internals)
+// Partition all nodes into clusters. The total number of clusters is read from
+// the `flagCollectionClusters` flag.
+// The number of nodes in each cluster is deterministic
+// and only depends on the number of clusters and nodes.
+// However, the list of nodes in each cluster if non-deterministic and produces
+// different lists on each function call.
+func constructClusterAssignment(participants flow.IdentityList) (flow.AssignmentList, flow.ClusterList) {
```

Comment:
background discussion about this change happened in another PR: https://github.com/onflow/flow-go/pull/4052#discussion_r1139380755